src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -227,15 +238,14 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
   }
 
   /**
-   * Error event handler.
+   * error event handler.
    * @param {Function} reject - The reject promise function.
    * @param {FakeEvent} error - The error fake event.
    * @private
    * @returns {void}
    */
   _onError(reject: Function, error: FakeEvent): void {
-    NativeAdapter._logger.error(error);
-    reject(error);
+    reject(new Error(Error.Severity.CRITICAL, Error.Category.PLAYER, Error.Code.NATIVE_ADAPTER_LOAD_FAILED, error.payload));
```

Comment:
You are already listening to error event and dispatching event in HTML5, I think you can just call reject and let the error handler fire the event, otherwise we will get two error events